components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -884,13 +884,13 @@ public static RevocationStatus getRevocationStatus(X509Certificate peerCert, X50
         if (authorityInformationAccess != null) {
             accessDescriptions = authorityInformationAccess.getAccessDescriptions();
             for (AccessDescription accessDescription : accessDescriptions) {
-
-                GeneralName gn = accessDescription.getAccessLocation();
-                if (gn.getTagNo() == GeneralName.uniformResourceIdentifier &&
-                        accessDescription.getAccessMethod().equals(X509ObjectIdentifiers.ocspAccessMethod)) {
-                    DERIA5String str = DERIA5String.getInstance(gn.getName());
-                    String accessLocation = str.getString();
-                    ocspUrlList.add(accessLocation);
+                if (X509ObjectIdentifiers.ocspAccessMethod.equals(accessDescription.getAccessMethod())) {
+                    GeneralName gn = accessDescription.getAccessLocation();
+                    if (gn.getTagNo() == GeneralName.uniformResourceIdentifier) {
```

Comment:
can `gn` be null?
can we use equals for `gn.getTagNo() == GeneralName.uniformResourceIdentifier`?

Reply:
Null check added in https://github.com/wso2-extensions/identity-x509-commons/pull/13/commits/335d3408181f632f5012157f26f80c39759b4ecf. Since this comparison is between two 'int' primitives, equals cannot be used.